aspnet/mvc/controllers/routing.rst

```diff
@@ -1,8 +1,140 @@
+
+.. code-block:: c#
+
+  [Route("category/{movie}.rss", Order = 0, Name = "MovieCategoryFeed")]
+  public IActionResult MovieRSS(string name)
+
+The above creates a route that directs to the ``MovieRSS`` method, and looks something like the following:
+
+.. code-block:: c#
+
+  /category/GoneWithTheWind.rss 
+
+You may create custom route templates if you can't find any other built-in solution. Generally, it is because you need to dynamically compute a route template. To do so, subclass any route attribute, or implement the ``IApplicationModelConvention`` interface. Keep the code in the custom attribute simple as a limited extensibility point.   
+
+Using Application Model to customize attribute routes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Implement the ``Apply`` method of the ``IApplicationModelConvention`` interface to create an attribute route. For more information on the MVC application model, see the documentation on `ApplicationModel <https://docs.asp.net/en/latest/mvc/controllers/application-model.html>`_.
+
+Action constraints
+------------------
+
+MVC provides a few ways to apply finer control over how it processes routes. For example, you might want to restrict URLs to a particular HTTP verb. Other times you might want to dynamically prepend or append strings to routes, such as "api" for routes that lead to an API action. Or perhaps you're working on a blog engine and want the publish date and SEO information in the URLs. Constraints are great for when you need to modify the default behavior of actions.
+    
+HTTP method constraints
+^^^^^^^^^^^^^^^^^^^^^^^
+
+MVC allows you to restrict routes so they only accept specific HTTP verbs. Below is a list of supported HTTP verb constraints: 
+
+- ``[HttpDelete]``
+- ``[HttpGet]``
+- ``[HttpHead]``
+- ``[HttpPatch]``
+- ``[HttpPost]``
+- ``[HttpPut]``
+
+If you want to accept only ``HTTP GET`` requests that pass an integer to the ``GetMovie`` action, apply the ``[HttpGet]`` constraint to the ``GetMovie``, as shown here:
+
+.. code-block:: c#
+
+  [HttpGet("{id}", Name = "GetMovie")]
+  public IActionResult GetMovie([FromRoute] int id)
+  
+The code also constrains the route so it accepts integers only. This means that the route ``/GetMovie/1337`` will work, but ``/GetMovie/GoneWithTheWind`` will not.  
+
+Custom Constraints
+^^^^^^^^^^^^^^^^^^
+  
+Constraints are configured by passing them as a parameter to the ``MapRoute`` method at startup, as shown in a previous sample. You can create a custom action constraint by implementing the ``IActionConstraint`` interface. It's an extensibility point to verify if an action can run for a given request. This enabes you to create a nice experience from the URL point of view in that you can keep your URLs short and simple.
```

Comment:
`IActionConstraint` is the extensibility point used by things like `[HttpGet(...)]` - it's part of the action setup in MVC (usually an attribute).

`IRouteConstraint` is a routing extensibility point used by things like `{id:int}`